src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java

```diff
@@ -0,0 +1,249 @@
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.hamcrest.Matcher;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.CheckInByBarcodeRequestBuilder;
+import api.support.builders.HoldingBuilder;
+import api.support.builders.ItemBuilder;
+import api.support.builders.NoticeConfigurationBuilder;
+import api.support.builders.NoticePolicyBuilder;
+import api.support.builders.RequestBuilder;
+import api.support.fixtures.ItemExamples;
+import api.support.fixtures.TemplateContextMatchers;
+import api.support.http.InventoryItemResource;
+import io.vertx.core.json.JsonObject;
+
+public class RequestScheduledNoticesProcessingTests extends APITests {
+
+  private UUID templateId = UUID.randomUUID();
+  private InventoryItemResource item;
+  private IndividualResource requester;
+  private IndividualResource pickupServicePoint;
+
+  @Before
+  public void beforeEach()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    ItemBuilder itemBuilder = ItemExamples.basedUponSmallAngryPlanet(
+      materialTypesFixture.book().getId(), loanTypesFixture.canCirculate().getId());
+    HoldingBuilder holdingBuilder = itemsFixture.applyCallNumberHoldings(
+      "CN",
+      "Prefix",
+      "Suffix",
+      singletonList("CopyNumbers"));
+
+    item = itemsFixture.basedUponSmallAngryPlanet(itemBuilder, holdingBuilder);
+    requester = usersFixture.steve();
+    pickupServicePoint = servicePointsFixture.cd1();
+  }
+
+  @Test
+  public void uponAtRequestExpirationNoticeShouldBeSentAndDeleted()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    JsonObject noticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withRequestExpirationEvent()
+      .withUponAtTiming()
+      .sendInRealTime(true)
+      .create();
+    setupNoticePolicyWithRequestNotice(noticeConfiguration);
+
+    LocalDate requestExpiration = LocalDate.now(UTC).minusDays(1);
+    IndividualResource request = requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(DateTime.now())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    scheduledNoticeProcessingClient.runRequestNoticesProcessing();
+    List<JsonObject> notices = patronNoticesClient.getAll();
+
+    assertThat(notices, hasSize(1));
+    assertThat(notices.get(0), getTemplateContextMatcher(templateId, request));
+  }
+
+  @Test
+  public void beforeRequestExpirationNoticeShouldBeSentAndDeleted()
```

Comment:
Under what circumstances should it be deleted, when it is first sent, because it is not recurring?

Reply:
Yes, because is not recurring. Extended method's name to reflect this.